enhancements/proposals/1665-metadata/README.md

```diff
@@ -0,0 +1,140 @@
+
+In case there is a conflict with managed metadata, the managed metadata will win.
+
+These changes are purely intended for metadata and shall not change the selectors used.
+
+### User Stories
+
+#### DataDog metrics
+As a user I want to configure [DataDog metrics](https://docs.datadoghq.com/containers/kubernetes/prometheus/?tab=kubernetesadv2#setup) for certain pods.
+This is done by adding `ad.datadoghq.com/<CONTAINER_IDENTIFIER>.checks` to Pod's annotations.
+
+#### Third Party Service and Ingress options
+As a user I want to configure additional options for my service or ingress.
+These third party options are usually controlled by adding annotations or labels, e.g.:
+- [`networking.gke.io/load-balancer-type: "Internal"` annotation for GKE](https://cloud.google.com/kubernetes-engine/docs/how-to/internal-load-balancing)
+- [`service.beta.kubernetes.io/aws-load-balancer-eip-allocations: eipalloc-xxxxxxxxxxxxxxxxx,eipalloc-yyyyyyyyyyyyyyyyy` annotation for assigning elastic IPs](https://docs.aws.amazon.com/eks/latest/userguide/network-load-balancing.html)
+- [Ingress Controller sharding using labels and selectors in OCP](https://docs.openshift.com/container-platform/4.14/networking/ingress-sharding.html)
+
+#### Cost tracking
+As a user I want to add a label to all of my resources and use it to track costs.
+
+#### PagerDuty
+As a user I want to add extra label to my workload to route dynamic PagerDuty alerts.
+
+#### PVC properties
+As a user I want to add extra annotations to configure additional properties for PVCs, like "ebs.csi.aws.com/iops" or "ebs.csi.aws.com/throughput".
+
+### Risks and Mitigations
+
+Not known.
+
+## Design Details
+
+### API changes for ScyllaClusters.scylla.scylladb.com/v1
+
+The following API changes show how we'd approximately extend `scyllaclusters.scylla.scylladb.com/v1.spec` specification. Only new fields are shown for existing structs.
+
+```golang
+package v1
+
+type ObjectTemplateMetadata struct {
+	// labels is a custom key value map that gets merged with managed object annotations.
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// annotations is a custom key value map that gets merged with managed object annotations.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+type ScyllaClusterSpec struct {
+	// podMetadata controls shared metadata for all pods created based on this spec.
+	// +optional
+	PodMetadata *ObjectTemplateMetadata `json:"podMetadata,omitempty"`
+
+	// ...
+}
+
+type RackSpec struct {
+	// volumeClaimMetadata controls shared metadata for the volume claim for this rack.
+	// At this point, this field is applied only for the initial claim and is not reconciled during its lifetime.
+	// Note that this may get fixed in the future and this behaviour shouldn't be relied on.
+	// +optional
+	VolumeClaimMetadata *ObjectTemplateMetadata `json:"volumeClaimMetadata,omitempty"`
```

Comment:
Did you consider placing this in `rackSpec.storage`?  E.g. `rackSpec.storage.capacity` specifies the requested size of that same PVC. It doesn't seem perfect either, buy maybe a bit more concise? Wdyt?

Reply:
I have but given how it's structured today it would kind of loose the metadata vs. spec separation. I kind of wish this API was modeled similarly to Kubernetes resources but here we are. There is a lot of space to make this better in v2.

Picking the less bad option - given the API has only `storage` field, we might rename `StorageSpec` struct to just `Storage` and put metadata struct into it. So it would be more like `storage.metadata.annotations`. Is that what you'd prefer?

Reply:
I don't have a strong preference, although 'storage.metadata.annotations' seems more intuitive to me. I guess pick whatever you think will be easier to transpile to v2.